docs/testing.md

```diff
@@ -30,9 +30,7 @@ Each of these tests could fail for different reasons; see "Possible errors" belo
 
 These tests run `git diff` for your current branch against some baseline branch to determine what sources are new or changed.
 
-Since it's impossible for us to accurately guess what the right baseline branch would be in your case (`origin/master`? `upstream/master`?), you will need to create a `gitdiff.json` in `tests/integration/shared/sources`.  See `gitdiff.json.example` in that folder for reference.
-
-If this file is missing, the test will stop with a giant warning message.  (In CI, we just use `origin/master` as the base branch, and this file isn't required).
+If your upstream branch is not named `origin/master`, you will need to create a `gitdiff.json` in `tests/integration/shared/sources`.  See `gitdiff.json.example` in that folder for reference.
```

Comment:
This seems fair enough, thanks!  I was trying to be more general, but I think being opinionated is good, especially for people following the usual workflow.  Honestly, even if they somehow get it wrong, it's not the end of the world!